climpred/classes.py

```diff
@@ -345,46 +544,21 @@ def bootstrap(
               https://doi.org/10/f4jjvf.
 
         """
-        is_initialized(self.control, 'control', 'a bootstrap')
-        # compute for single variable.
-        if var is not None:
-            return bootstrap_perfect_model(
-                self.initialized[var],
-                self.control[var],
-                metric=metric,
-                comparison=comparison,
-                sig=sig,
-                bootstrap=bootstrap,
-                pers_sig=pers_sig,
-            )
-        # compute for all variables in control.
-        else:
-            if len(self.initialized.data_vars) == 1:
-                for var in self.initialized.data_vars:
-                    var = var
-                return bootstrap_perfect_model(
-                    self.initialized[var],
-                    self.control[var],
-                    metric=metric,
-                    comparison=comparison,
-                    sig=sig,
-                    bootstrap=bootstrap,
-                    pers_sig=pers_sig,
-                )
-            else:
-                boot = {}
-                for var in self.control.data_vars:
-                    res = bootstrap_perfect_model(
-                        self.initialized[var],
-                        self.control[var],
-                        metric=metric,
-                        comparison=comparison,
-                        sig=sig,
-                        bootstrap=bootstrap,
-                        pers_sig=pers_sig,
-                    )
-                    boot[var] = res
-                return boot
+        has_dataset(self._datasets['control'], 'control', 'bootstrap')
+        input_dict = {
+            'ensemble': self._datasets['initialized'],
+            'control': self._datasets['control'],
+            'init': True,
+        }
+        return self._apply_climpred_function(
+            bootstrap_perfect_model,
+            input_dict=input_dict,
+            metric=metric,
+            comparison=comparison,
+            sig=sig,
+            bootstrap=bootstrap,
+            pers_sig=pers_sig,
+        )
```

Comment:
Neat, somehow compressed all that into a lot less lines!

Reply:
I just got rid of the `var` keyword. As @aaronspring pointed out, the user can subselect those variables if they only want certain ones applied.